crates/bevy_pbr/src/bloom/mod.rs

```diff
@@ -0,0 +1,769 @@
+use std::num::NonZeroU32;
+
+use bevy_app::{App, Plugin};
+use bevy_asset::{load_internal_asset, HandleUntyped};
+use bevy_core_pipeline::{
+    fullscreen_vertex_shader::fullscreen_shader_vertex_state, prelude::Camera3d,
+};
+use bevy_ecs::{
+    prelude::{Component, Entity},
+    query::{QueryState, With},
+    system::{Commands, Query, Res, ResMut, Resource},
+    world::{FromWorld, World},
+};
+use bevy_math::UVec2;
+use bevy_reflect::TypeUuid;
+use bevy_render::{
+    camera::ExtractedCamera,
+    prelude::Camera,
+    render_graph::{Node, NodeRunError, RenderGraph, RenderGraphContext, SlotInfo, SlotType},
+    render_phase::TrackedRenderPass,
+    render_resource::{
+        AddressMode, BindGroup, BindGroupDescriptor, BindGroupEntry, BindGroupLayout,
+        BindGroupLayoutDescriptor, BindGroupLayoutEntry, BindingResource, BindingType,
+        BlendComponent, BlendFactor, BlendOperation, BlendState, BufferBindingType,
+        CachedRenderPipelineId, ColorTargetState, ColorWrites, DynamicUniformBuffer, Extent3d,
+        FilterMode, FragmentState, LoadOp, MultisampleState, Operations, PipelineCache,
+        PrimitiveState, RenderPassColorAttachment, RenderPassDescriptor, RenderPipelineDescriptor,
+        SamplerBindingType, SamplerDescriptor, Shader, ShaderStages, ShaderType, TextureDescriptor,
+        TextureDimension, TextureSampleType, TextureUsages, TextureView, TextureViewDescriptor,
+        TextureViewDimension,
+    },
+    renderer::{RenderContext, RenderDevice, RenderQueue},
+    texture::{CachedTexture, TextureCache},
+    view::ViewTarget,
+    Extract, RenderApp, RenderStage,
+};
+use bevy_utils::HashMap;
+
+pub mod draw_3d_graph {
+    pub mod node {
+        /// Label for the bloom render node.
+        pub const BLOOM: &str = "bloom";
+    }
+}
+
+const BLOOM_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 929599476923908);
+
+pub struct BloomPlugin;
```

Comment:
Am I correct in thinking that this should also work in 2d? If that's the case, I wonder if we could share the plugin for 2d and 3d. It might make sense to move this to bevy_core_pipeline if that's the case and extract the node to it's own file. A bit like the tonemapping and upscaling.

Reply:
Oh, hmm I didn't think of 2d. It should work in 2d I think, I would just need to remove the Camera3d filter. 

Should we make a bevy_post_process crate or something similar? I'm not sure if AA, tonemapping, upscaling, bloom, etc belong in either bevy_pbr/bevy_sprite or bevy_core_pipeline.

Reply:
I think, for this PR, keep it in core_pipeline, but we should probably have a new crate in the future.

Reply:
I think it would also need to be added to the `bevy_core_pipeline::core_2d` graph.

Reply:
Did we decide not to do 2D? Why?